config.py

```diff
@@ -1395,14 +1398,18 @@ def get_bids_root() -> pathlib.Path:
                'root folder of your BIDS dataset')
         raise ValueError(msg)
 
-    return pathlib.Path(bids_root).expanduser()
+    return (pathlib.Path(bids_root)
+            .expanduser()
+            .resolve(strict=True))
 
 
 def get_deriv_root() -> pathlib.Path:
     if deriv_root is None:
         return get_bids_root() / 'derivatives' / PIPELINE_NAME
     else:
-        return pathlib.Path(deriv_root).expanduser()
+        return (pathlib.Path(deriv_root)
+                .expanduser()
+                .resolve())
```

Comment:
no `strict` here because it's okay if this directory doesn't exist, as we'll then just create it